src/main/java/com/github/manolo8/darkbot/core/manager/GuiManager.java

```diff
@@ -121,6 +121,8 @@ private void tryRevive() {
             API.writeMemoryLong(repairAddress + 32, main.config.GENERAL.SAFETY.REVIVE_LOCATION);
             API.mouseClick(MapManager.clientWidth / 2, (MapManager.clientHeight / 2) + 190);
             repairTime = System.currentTimeMillis();
+
+            new HangarManager(this.main).checkDrones();
```

Comment:
Don't create the hangar manager instance here, create it inside the BackpageManager. Also, we don't want to check for drones synchronously, this will be a big issue because it's a blocking operation. It needs to be done in the backpage thread. You should set some flag for drones to be checked over in the backpage manager, on the next thread cycle.